### ISSUE TITLE
LRG Gaussian random field mocks need to be updated

### ISSUE THREAD

**Issue by moustakas** [desihub/desitarget#263]
@dkirkby @fjaviersanchez @forero 

The current version of the Gaussian random field mocks for LRGs are too undersampled -- and likely have the incorrect b(z) and n(z) -- for the current LRG selection.  Recall that the selection was changed from 350/deg2 (in the Final Design Report) to 500/deg2 in the current, "retuned" selection (see [here](https://desi.lbl.gov/trac/wiki/TargetSelectionWG/TargetSelection#LuminousRedGalaxiesLRG)).

In the `select_mock_targets` refactor (#261) I can't achieve the desired density of 500/deg2 after target selection is applied to the noisy (simulated) photometry with the current LRG mock -- I need more candidate objects.

But in addition, the new LRG selection is choosing a different population than the FDR selection -- the mean bias is likely lower than what had been assumed previously and n(z) is definitely different -- so that needs to be updated as well.

**Comment by dkirkby** [comment]
It looks like the [desimodel n(z)](https://desi.lbl.gov/svn/code/desimodel/trunk/data/targets/nz_lrg.dat) has already been updated to 500/sq.deg.  Is there also a new baseline b(z) we should use?  We currently assume b(z)D(z) = 1.70 from the FDR.

**Comment by moustakas** [comment]
Thanks @dkirkby I've posed this question to the Targeting WG.

**Comment by moustakas** [comment]
Based on the discussion on-list, it looks like this is still a reasonable estimate of the bias for this sample, @dkirkby.  

Also, the desired LRG target density is 480/deg2, so having the mock oversampled by a factor of 2-4 would be very helpful.

**Comment by fjaviersanchez** [comment]
@moustakas should I use the N(z) from the desimodel or a different one? What about the b(z), do you prefer that I still use the same or is it better if I use a new value for b(z)*D(z)?

**Comment by moustakas** [comment]
@rongpu could you double-check that the n(z) for LRGs in desimodel matches your current predictions, i.e., the n(z) shown in Figure 4.7 of the late-2017 target selection document, and update it if necessary?
https://desi.lbl.gov/svn/code/desimodel/trunk/data/targets/nz_lrg.dat

@fjaviersanchez as for the bias---based on on-list discussions we should keep the same bias assumed previously, `b(z) = 1.70/D(z)`

**Comment by rongpu** [comment]
I've computed the **surface density** n(z) that corresponds to Figure 4.7 in the [updated TS document](https://desi.lbl.gov/DocDB/cgi-bin/private/RetrieveFile?docid=3349;filename=TS_2017.pdf;version=4), and I've updated the N_lrg column in the nz_lrg.dat file on the SVN.

When comparing the updated n(z) to the previous n(z) from nz_lrg.dat (see plot below), I noticed that the the previous surface density n(z) might have been incorrectly computed from the comoving number densities on the [LRG retuning page](https://desi.lbl.gov/trac/attachment/wiki/TargetSelectionWG/LRGretuning/), resulting an overestimation at lower redshift. That could affect the mocks using that version of n(z).

Since I'm using photo-z's  and thus have a much larger sample than previous analysis using AGES & VIPERS spec-z's, I can provide the n(z) with much finer redshift bins if needed.

![image](https://user-images.githubusercontent.com/16765087/35408116-f1e56fa6-01db-11e8-978e-99ba20f405e7.png)

**Comment by moustakas** [comment]
Yes, please.

**Comment by fjaviersanchez** [comment]
@moustakas Please find the latest version of the mocks here: `/global/project/projectdirs/desi/mocks/GaussianRandomField/v0.0.7_2LPT`. This is what the LRGs look like right now:

<img width="694" alt="screen shot 2018-01-28 at 11 07 56 am" src="https://user-images.githubusercontent.com/13971732/35485954-b0cba340-041b-11e8-98ea-dc9458d3a996.png">

The average density of the non-zero pixels is 942 galaxies/sq-deg before selection. Let me know if this works for you or if you would like a version with a larger density.

Also, this version uses 2LPT instead of a lognormal approximation so the smaller scales should be more realistic.

**Comment by fjaviersanchez** [comment]
@moustakas @forero have you had time to check the new version of the mocks? Do you need something else?

**Comment by moustakas** [comment]
Thanks @fjaviersanchez, I'm testing these now.  Just one clarification: it looks like the ELG and QSO mock catalogs been updated as well, but other than the change in small-scale clustering (via the use of the LPT vs log-normal algorithm) these catalogs should not be different than the `v0.0.5` ones, correct?

**Comment by moustakas** [comment]
Thanks @fjaviersanchez.  Here's what I get for the spatial and redshift distributions.  I wonder if we shouldn't increase the density of the QSO mocks because I'm having trouble hitting our goal (uncontaminated) target density of 120/deg2 after target selection cuts have been applied.  Alternatively, we could just move forward with these for now.

![qso](https://user-images.githubusercontent.com/1431820/36078215-ba641cf8-0f41-11e8-9652-6bbcd3f94344.png)

![lrg](https://user-images.githubusercontent.com/1431820/36080810-32162922-0f64-11e8-9a7d-c5c5580f36df.png)

![elg](https://user-images.githubusercontent.com/1431820/36080894-88c58262-0f65-11e8-825b-96cbcbe841a5.png)

For the record here's what I (roughly) did to make one of these plots:
```python
from desitarget.mock.mockmaker import QSOMaker
qq = QSOMaker()
qdata = qq.read()
qq.qamock_sky(qdata, png='qso.png')
```

**Comment by fjaviersanchez** [comment]
@moustakas I can multiply the current N(z) times 1.5 for QSOs if that helps!

**Comment by fjaviersanchez** [comment]
And yes, other than the small-scale clustering there should be no change in ELGs and QSOs.

**Comment by moustakas** [comment]
Apologies for closing this prematurely.  Yes, if it's trivial to increase the density of the QSO mock I think that would help.

**Comment by fjaviersanchez** [comment]
Yes, it's pretty easy. I'm going to run it now. I'll get back to you once I have the new set of mocks.

**Comment by fjaviersanchez** [comment]
@moustakas I updated the mocks in the same folder as before with a larger number density of QSOs. The other files should be the same. I tried to use the commands that you listed above but they don't seem to work with the version of desitarget at NERSC (I pulled the master branch version but I think that the code is in the `fix-densities` branch). I'll try to install it and make the plots that you showed above too but feel free to play with this version, and please let me know if these are fine or you prefer a larger number density.
